torchdynamo/eval_frame.py

```diff
@@ -556,6 +557,16 @@ def graph_with_interpreter(*args):
     return (new_graph, out_guards)
 
 
+def assume_constant_result(fn):
+    setattr(fn, "DYNAMO_MARKED_CONSTANT", True)
+    # @functools.wraps(fn)
+    # def inner(*args, **kwargs):
+    #     return fn(*args, **kwargs)
+    if config.fake_tensor_propagation:
+        unimplemented("Constant result capture is not supported with fake tensors.")
```

Comment:
`unimplemented` it meant to be raised while symbolically evaluating, not from user facing APIS.

Let's raise an exception directly here.

Reply:
Yeah, that's a good point.